paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/SchemaChangeITCase.java

```diff
@@ -28,7 +32,211 @@
 /** ITCase for schema changes. */
 public class SchemaChangeITCase extends CatalogITCaseBase {
 
-    // TODO cover more cases once Flink supports more ALTER operations.
+    @Test
+    public void testAddColumn() {
+        sql("CREATE TABLE T (a STRING, b DOUBLE, c FLOAT)");
+        sql("INSERT INTO T VALUES('aaa', 1.2, 3.4)");
+        sql("ALTER TABLE T ADD d INT");
+        List<Row> result = sql("SHOW CREATE TABLE T");
+        assertThat(result.toString())
+                .contains(
+                        "CREATE TABLE `PAIMON`.`default`.`T` (\n"
+                                + "  `a` VARCHAR(2147483647),\n"
+                                + "  `b` DOUBLE,\n"
+                                + "  `c` FLOAT,\n"
+                                + "  `d` INT\n"
+                                + ")");
+        sql("INSERT INTO T VALUES('bbb', 4.5, 5.6, 5)");
+        result = sql("SELECT * FROM T");
+        assertThat(result.toString()).isEqualTo("[+I[aaa, 1.2, 3.4, null], +I[bbb, 4.5, 5.6, 5]]");
+
+        // add column with after position
+        sql("CREATE TABLE T1 (a STRING, b DOUBLE, c FLOAT)");
+        sql("INSERT INTO T1 VALUES('aaa', 1.2, 3.4)");
+        sql("ALTER TABLE T1 ADD d INT AFTER b");
+        result = sql("SHOW CREATE TABLE T1");
+        assertThat(result.toString())
+                .contains(
+                        "CREATE TABLE `PAIMON`.`default`.`T1` (\n"
+                                + "  `a` VARCHAR(2147483647),\n"
+                                + "  `b` DOUBLE,\n"
+                                + "  `d` INT,\n"
+                                + "  `c` FLOAT\n"
+                                + ")");
+        sql("INSERT INTO T1 VALUES('bbb', 4.5, 6, 5.6)");
+        result = sql("SELECT * FROM T1");
+        assertThat(result.toString()).isEqualTo("[+I[aaa, 1.2, null, 3.4], +I[bbb, 4.5, 6, 5.6]]");
+
+        // add column with first position
+        sql("CREATE TABLE T2 (a STRING, b DOUBLE, c FLOAT)");
+        sql("INSERT INTO T2 VALUES('aaa', 1.2, 3.4)");
+        sql("ALTER TABLE T2 ADD d STRING FIRST");
+        result = sql("SHOW CREATE TABLE T2");
+        assertThat(result.toString())
+                .contains(
+                        "CREATE TABLE `PAIMON`.`default`.`T2` (\n"
+                                + "  `d` VARCHAR(2147483647),\n"
+                                + "  `a` VARCHAR(2147483647),\n"
+                                + "  `b` DOUBLE,\n"
+                                + "  `c` FLOAT\n"
+                                + ")");
+
+        sql("INSERT INTO T2 VALUES('paimon', 'bbb', 3.4, 6.7)");
+        result = sql("SELECT * FROM T2");
+        assertThat(result.toString())
+                .isEqualTo("[+I[null, aaa, 1.2, 3.4], +I[paimon, bbb, 3.4, 6.7]]");
+    }
+
+    @Test
+    public void testDropColumn() {
```

Comment:
1. Test: insert value before and read after.
2. Please add test for drop partition key and primary key (should throw exception). See `SparkSchemaEvolutionITCase`.